include/net/net_pkt.h

```diff
@@ -2007,6 +2010,156 @@ const char *net_pkt_pool2str(struct net_buf_pool *pool);
 #define net_pkt_print(...)
 #endif /* CONFIG_NET_PKT_LOG_LEVEL >= LOG_LEVEL_DBG */
 
+/* New allocator, and API are defined below.
+ * This will be simpler when time will come to get rid of former API above.
+ */
+#if CONFIG_NET_PKT_LOG_LEVEL >= LOG_LEVEL_DBG
+
+struct net_pkt *net_pkt_allocate_debug(s32_t timeout,
+				       const char *caller, int line);
+#define net_pkt_allocate(_timeout)					\
```

Comment:
What about making the function name a bit shorter so I propose we change this to `net_pkt_alloc()`

Reply:
true. I had that name at first, I don't know why I went for the plain allocate word.